dspace-api/src/main/java/org/dspace/importer/external/elsevier/UpdateElsevierItems.java

```diff
@@ -0,0 +1,297 @@
+                        updatePermissions(context, item);
+                    }
+
+                    if (line.hasOption("a")) {
+                        printAndLog("checking pii updates for item " + item.getHandle());
+                        assignPii(context, item);
+                    }
+
+                    if (line.hasOption("m")) {
+                        printAndLog("checking metadata updates for item " + item.getHandle());
+                        importMetadata(context, item);
+                    }
+
+                    itemService.update(context,item);
+                    context.dispatchEvents();
+                }
+
+                context.restoreAuthSystemState();
+                context.complete();
+            }
+            else {
+                printAndLog("no items found.");
+            }
+        }
+        catch(Exception e){
+            System.out.println(e.getMessage());
+        }finally{
+            if (context != null) {
+                context.abort();
+            }
+        }
+    }
+
+    private static Options CreateCommandLineOptions() {
+        Options options = new Options();
+        Option testOption = OptionBuilder.withArgName("test").withDescription("output changes without applying them").create('t');
+        Option permissionOption = OptionBuilder.withArgName("permissions").withDescription("Update the item permissions").create('p');
+        Option piiOption = OptionBuilder.withArgName("assignpii").withDescription("Lookups uo the pii for items with a doi (but no pii) and add it to the metadata").create('a');
+        Option metadataOption = OptionBuilder.withArgName("metadata").withDescription("import the item metadata").create('m');
+        Option forceOption = OptionBuilder.withArgName("force").withDescription("force update changes from elsevier").create('f');
+        Option identifierOption = OptionBuilder.withArgName("item").hasArg().withDescription("specify a handle to update a single item").create('i');
+
+        options.addOption(testOption);
+        options.addOption(permissionOption);
+        options.addOption(piiOption);
+        options.addOption(metadataOption);
+        options.addOption(forceOption);
+        options.addOption(identifierOption);
+        return options;
+    }
+
+    private static void updatePermissions(Context context, Item item) throws SQLException, AuthorizeException {
+        List<Bundle> bundles = item.getBundles();
+
+        for (Bundle bundle : bundles) {
+            List<Bitstream> bitstreams = bundle.getBitstreams();
+
+            for (Bitstream bitstream : bitstreams) {
+                boolean identical = fileAccessFromMetadataService.fileAccessIdentical(context, bitstream);
+
+                if(!identical){
+                    boolean overruled = Boolean.parseBoolean(bitstreamService.getMetadata(bitstream, "workflow.fileaccess.overruled"));
```

Comment:
How does this work? The way I read that code, every bitstream that does not have a boolean metadata "workflow.fileaccess.overruled" might be changed, if this command line tool runs with the option `-p` and without the option `-i`. What do I miss?